docs/generate-time-series/javascript.rst

```diff
@@ -0,0 +1,232 @@
+
+    > var query = `
+    ...     CREATE TABLE load (
+    ...         timestamp TIMESTAMP GENERATED ALWAYS AS CURRENT_TIMESTAMP, 
+    ...         avg_1m REAL, 
+    ...         avg_5m REAL, 
+    ...         avg_15m REAL)`
+    > await client.query(query)
+
+
+In the `CrateDB Admin UI`_, you should see the new table when you navigate to
+the *Tables* screen using the left-hand navigation menu:
+
+.. image:: ../_assets/img/generate-time-series/table.png
+
+
+Record your system load
+=======================
+
+With the table in place, you can start recording load averages.
+
+The following command calls `os.loadavg`_ and uses the result triple as `input
+values`_ for the `INSERT`_ query::
+
+    > await client.query("INSERT INTO load (avg_1m, avg_5m, avg_15m) VALUES (?, ?, ?)", os.loadavg())
+
+Press the up arrow on your keyboard and hit *Enter* to run the same command a
+few more times.
+
+When you're done, you can `SELECT`_ that data back out of CrateDB, like so::
+
+    > client.query('SELECT * FROM load ORDER BY timestamp DESC', (err, res) => {
+    ...   if (err) {
+    ...       console.log(err.stack)
+    ...     } else {
+    ...       console.log(res.rows)
+    ...     }
+    ... })
+    [ { timestamp: 2020-02-13T14:23:23.566Z,
+        avg_1m: 0.83251953,
+        avg_5m: 0.54785156,
+        avg_15m: 0.2265625 },
+      { timestamp: 2020-02-13T14:23:22.968Z,
+        avg_1m: 0.90527344,
+       avg_5m: 0.5571289,
+        avg_15m: 0.22802734 },
+      { timestamp: 2020-02-13T14:23:21.144Z,
+        avg_1m: 0.90527344,
+        avg_5m: 0.5571289,
+        avg_15m: 0.22802734 } ]
+
+Here we have recorded three sets of load averages with a corresponding
+timestamp. We can now close the connection::
+
+    > await client.end()
+
+
+Automate it
+===========
+
+Now we have the basics figured out, let's automate the data collection.
+
+Our approach will change a little bit here. We were previously using a `Client`_ to
+connect to and insert data into CrateDB. However, Clients are ephemeral, and once closed
+they need to be recreated. However, creating a new Client requires a handshake with CrateDB, and
+this overhead cost can be prohibative if we are creating new clients rapidly.
+
+Instead, we can use a `Connection Pool`_ to manage our connections. Create a new file called
+``monitor-load.js``, like this:
+
+.. code-block:: javascript
+
+    const os = require('os')
+    const { Pool } = require('pg')
+    const pool = new Pool({
+        user: 'crate',
+        host: 'localhost',
+        database: 'doc',
+        port: 5432,
+    })
+
+    function insert() {
+        pool.query(
+            "INSERT INTO load (avg_1m, avg_5m, avg_15m) VALUES (?, ?, ?)",
+            os.loadavg(),
+            (err, res) => {
+                if (err) {
+                    console.log(err.stack)
+                } else {
+                    console.log("INSERT OK")
+                }
+            }
```

Comment:
should we add a note about adding exception handling?

also: should we add this to the python example?

or, alternatively, should we strip this demo down so it's as basic as the python one with no exception handling at all. the downside of this is that the script is brittle for actual real-world use and we pass up the opportunity to introduce exception handling as a tool. the upside is: it keeps the tutorial simple and means that the end example isn't using too much extra code that hasn't been used in the repl sections

Reply:
Hm, I'm torn. On the one hand, without error handling the above could be written as a much simpler:

```
await pool.query("INSERT INTO load (avg_1m, avg_5m, avg_15m) VALUES (?, ?, ?)", os.loadavg())
console.log("INSERT OK")
```

But, on the other hand, I would also be tetchy about a tutorial that says, explicitly, "INSERT OK" without actually ensuring that the insert was ok. I think adding a little bit of exception handling to the python section is probably the best thing to do here. What do you think?

Reply:
okay yes, you've convinced me. :D let's keep it, and let's patch the python tutorial afterward. do you want to do that or should I?

Reply:
I'll take care of it!